diff --git a/soup/Socket.cpp b/soup/Socket.cpp
--- a/soup/Socket.cpp
+++ b/soup/Socket.cpp
@@ -1032,0 +1033,8 @@
+				for (const auto cipher_suite : hello.cipher_suites)
+				{
+					if (cipher_suite == TlsCipherSuite::TLS_EMPTY_RENEGOTIATION_INFO_SCSV)
+					{
+						client_supports_secure_renegotiation = true;
+						break;
+					}
+				}
diff --git a/soup/TlsCipherSuite.hpp b/soup/TlsCipherSuite.hpp
--- a/soup/TlsCipherSuite.hpp
+++ b/soup/TlsCipherSuite.hpp
@@ -29,0 +30,2 @@
+		TLS_EMPTY_RENEGOTIATION_INFO_SCSV = 0x00FF,
+
